Jenkinsfile.trigger

```
@@ -32,33 +36,47 @@ node {
 				[$class: 'RelativeTargetDirectory', relativeTargetDir: 'meta'],
 			],
 		))
+		pastFailedJobsJson = sh(returnStdout: true, script: '''#!/usr/bin/env bash
+			set -Eeuo pipefail -x
+
+			if ! json="$(wget -qO- "$JOB_URL/lastSuccessfulBuild/artifact/pastFailedJobs.json")"; then
+				echo >&2 'failed to get pastFailedJobs.json'
+				json='{}'
+			fi
+			jq <<<"$json" '.'
+		''').trim()
 	}
 
 	dir('meta') {
-		def queueJSON = ''
+		def queueJson = ''
 		stage('Queue') {
-			// TODO this job should export a JSON file that includes the number of attempts so far per failing buildId, and then this list should inject those values, initialize missing to 0, and sort by attempts so that failing builds always live at the bottom of the queue
-			queueJSON = sh(returnStdout: true, script: '''
-				jq -L.scripts '
-					include "meta";
-					[
-						.[]
-						| select(
-							needs_build
-							and (
-								.build.arch as $arch
-								| if env.BASHBREW_ARCH == "gha" then
-									[ "amd64", "i386", "windows-amd64" ]
-								else [ env.BASHBREW_ARCH ] end
-								| index($arch)
+			withEnv([
+				'pastFailedJobsJson=' + pastFailedJobsJson,
+			]) {
+				// using pastFailedJobsJson, sort the needs_build queue so that failing builds always live at the bottom of the queue
+				queueJson = sh(returnStdout: true, script: '''
+					jq -L.scripts '
+						include "meta";
+						(env.pastFailedJobsJson | fromjson) as $pastFailedJobs |
+						[
+							.[]
+							| select(
+								needs_build
+								and (
+									.build.arch as $arch
+									| if env.BASHBREW_ARCH == "gha" then
+										[ "amd64", "i386", "windows-amd64" ]
+									else [ env.BASHBREW_ARCH ] end
+									| index($arch)
+								)
 							)
-						)
-					]
-				' builds.json
-			''').trim()
+						] | sort_by($pastFailedJobs[.buildId].count // 0)
```

Just to make sure:
```console
$ jq -n '{} | .foo.bar // "baz"'
"baz"
```
:heart:

---

Oh, but maybe we should include some bit of my old `TODO` comment here, to remind us what this is doing / for?

```suggestion
						]
						# this Jenkins job exports a JSON file that includes the number of attempts so far per failing buildId so that this can sort by attempts which means failing builds always live at the bottom of the queue (sorted by the number of times they have failed, so the most failing is always last)
						| sort_by($pastFailedJobs[.buildId].count // 0)
```